lib/msal-core/src/UserAgentApplication.ts

```diff
@@ -1239,41 +1290,50 @@ export class UserAgentApplication {
    * @param {string} [hash=window.location.hash] - Hash fragment of Url.
    * @hidden
    */
-  private processCallBack(hash: string): void {
+  private processCallBack(hash: string, stateInfo: ResponseStateInfo, parentCallback?: Function): void {
     this.logger.info("Processing the callback from redirect response");
+    // get the state info from the hash
+    if (!stateInfo) {
+      stateInfo = this.getResponseState(hash);
+    }
 
-    // get the request Info from the hash
-    // TODO: Refactor this or saveTokenFromHash to return an AuthResponse object
-    const requestInfo = this.getRequestInfo(hash);
-
+    let response : AuthResponse;
+    let authErr : AuthError;
     // Save the token info from the hash
-    this.saveTokenFromHash(requestInfo);
-
-    const token = requestInfo.parameters[Constants.accessToken] || requestInfo.parameters[Constants.idToken];
-    const errorDesc = requestInfo.parameters[Constants.errorDescription];
-    const error = requestInfo.parameters[Constants.error];
+    try {
+      response = this.saveTokenFromHash(hash, stateInfo);
+    } catch (err) {
+      authErr = err;
+    }
 
-    let tokenType: string;
-    if (requestInfo.parameters[Constants.accessToken]) {
-      tokenType = Constants.accessToken;
+    if ((stateInfo.requestType === Constants.renewToken) && response.getAccessToken()) {
```

Comment:
This doesn't look correct, let's double check

Reply:
stateInfo.requestType is set when we check the state. If it matches the login state, we set requestType to Constants.login. If it matches the acquireToken state, then we set requestType to renewToken. I believe this check is correct, but I have changed the predicate to OR for the time being.